### PR TITLE
standardize Makefile for operator-sdk

### PR DESCRIPTION
**Pull request by yazug**
copy operator-sdk related stuff from skeleton
generated by operator-sdk init

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -248,13 +250,25 @@ $(KUTTL): $(LOCALBIN)
 	chmod +x $(LOCALBIN)/kubectl-kuttl
 
 .PHONY: operator-sdk
-operator-sdk: $(OPERATOR_SDK) ## Download operator-sdk locally if necessary.
-$(OPERATOR_SDK): $(LOCALBIN)
-	test -s $(OPERATOR_SDK) || curl -o $(LOCALBIN)/operator-sdk -L https://github.com/operator-framework/operator-sdk/releases/download/$(OPERATOR_SDK_VERSION)/operator-sdk_linux_amd64 && chmod +x $(LOCALBIN)/operator-sdk
+OPERATOR_SDK ?= $(LOCALBIN)/operator-sdk
+operator-sdk: ## Download operator-sdk locally if necessary.
+ifeq (,$(wildcard $(OPERATOR_SDK)))
```
Comment (rabi) [inline Makefile:255]: Why're we using wildcard here? Isn't the binary always named as operator-sdk or whatever name in $OPERATOR_SDK?

Comment (yazug) [reply Makefile:255]: This is the current Makefile from operator-sdk skeleton.  We are bringing all of the operators up to the same template as there have been some divergence.

As to why the wildcard I am not sure.  we would have to look into the operator-sdk changes that introduced that for the skeleton Makefile

Comment (rabi) [reply Makefile:255]: which skeleton Makefile we' are talking about here?  I guess it's something we've written.

Comment (yazug) [reply Makefile:255]: the skeleton Makefile that is generated if you run "operator-sdk init ... "  as if you were creating a new operator from scratch.  It generates a skeleton operator including a Makefile for building/testing and what I synced in with this patch brings our current Makefile closer to the latest generation as well as bring in some automation standardization improvements in the Makefile.  It allows the Makefile to better pick up or use provided OPERATOR_SDK var as well as several others but also just work for local dev workflow as well.

Comment (rabi) [reply Makefile:255]: > It allows the Makefile to better pick up or use provided OPERATOR_SDK var

Though I don't see any reason why we should look for sdk binary with wildcard function, if this what is generated by operator-sdk we can go with it.

**Review by rabi — APPROVED** [review]
/lgtm

File: Makefile (hunk 1 above)
Comment (yazug) [reply Makefile:255]: This construct allows the Makefile to check to see if first $(OPERATOR_SDK) is present locally in the filesystem, then it checks to see if operator-sdk is present (update OPERATPR_SDK to which operator-sdk),  if it is not present then download the cli and use that.

Comment (rabi) [reply Makefile:255]: I understand that but you don't use a wildcard to look if a file `$(OPERATOR_SDK) ` exists :smile:

**Pull request merged**